boa/src/value/mod.rs

```diff
@@ -158,13 +158,17 @@ impl Value {
     }
 
     /// Returns a new empty object
-    pub fn new_object(global: Option<&Value>) -> Self {
+    pub fn new_object(global: Option<&Value>, context: &Context) -> Self {
         let _timer = BoaProfiler::global().start_event("new_object", "value");
 
-        if let Some(global) = global {
-            let object_prototype = global.get_field("Object").get_field(PROTOTYPE);
-
-            let object = Object::create(object_prototype);
+        if global.is_some() {
+            let object = Object::create(
+                context
+                    .standard_objects()
+                    .object_object()
+                    .prototype()
+                    .into(),
+            );
```

Comment:
This function becomes a bit weird, it receives an `Option` that is used as a boolean (the content is never checked), I think it should be refactored to possably just receive a boolean. @HalidOdat what do you think?
I'm ok with it staying as is for this PR, just noting that it could be cleaned up.

Reply:
We could probably even remove the `global` argument. It is used with value `None` only in `Json::parse`:

https://github.com/boa-dev/boa/blob/22f8c5d0cee82262596b374832203be851d60525/boa/src/builtins/json/mod.rs#L72-L83

Maybe the call to `Value::new_object` could be replaced by `Value::object(Object::default())` in this location, or maybe it could even use an object with prototype as defined by default in `Value::new_object`

Reply:
> Maybe the call to `Value::new_object` could be replaced by `Value::object(Object::default())` in this location, or maybe it could even use an object with prototype as defined by default in `Value::new_object`

Yes. I think we should take this approach. we replace the calls where None is passed to `Value::object(Object::default())`.

Then remove the `global: Option<Value>` so it only takes `context`:
```rust
    pub fn new_object(context: &mut Context) -> Self {
        let _timer = BoaProfiler::global().start_event("new_object", "value");

        context.construct_object().into()
    }
```

In the future we should remove it entirely, but since it is used in many places we can begin the process of deprecation.